OpenEdXMobile/src/main/java/org/edx/mobile/util/UiUtils.kt

```diff
@@ -0,0 +1,244 @@
+import android.view.View
+import android.view.ViewGroup.MarginLayoutParams
+import android.view.animation.LinearInterpolator
+import android.view.animation.RotateAnimation
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.DrawableRes
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import androidx.viewpager2.widget.ViewPager2
+import org.edx.mobile.R
+import org.edx.mobile.base.MainApplication
+import org.edx.mobile.logger.Logger
+import org.edx.mobile.view.custom.SingleScrollDirectionEnforcer
+
+object UiUtils {
+    private val TAG = UiUtils::class.java.canonicalName
+    private val logger = Logger(UiUtils::class.java)
+
+    /**
+     * This function is used to return the passed Value in Display Metrics form
+     *
+     * @param point width/height as int
+     * @return float
+     */
+    fun getParamsInDP(r: Resources, point: Int): Float {
+        try {
+            return TypedValue.applyDimension(
+                TypedValue.COMPLEX_UNIT_DIP, point.toFloat(), r.displayMetrics
+            )
+        } catch (e: Exception) {
+            logger.error(e)
+        }
+        return 0F
+    }
+
+
+    fun dpToPx(context: Context, dpValue: Float): Float {
+        val scale = context.resources.displayMetrics.density
+        return dpValue * scale + 0.5f
+    }
+
+    /**
+     * Utility method to check the screen direction
+     *
+     * @return true if direction is LTR false else wise
+     */
+    fun isDirectionLeftToRight(): Boolean {
+        val config = MainApplication.instance().resources.configuration
+        return config.layoutDirection == View.LAYOUT_DIRECTION_LTR
+    }
+
+    /**
+     * CardView adds extra padding on pre-lollipop devices for shadows
+     * This function removes that extra padding from its margins
+     *
+     * @param cardView The CardView that needs adjustments
+     * @return float
+     */
+    fun adjustCardViewMargins(cardView: View) {
+        val params = cardView.layoutParams as MarginLayoutParams
+        params.topMargin -= cardView.paddingTop
+        params.leftMargin -= cardView.paddingStart
+        params.rightMargin -= cardView.paddingEnd
+        cardView.layoutParams = params
+    }
+
+    fun getDrawable(context: Context, @DrawableRes iconResId: Int, resSize: Int) =
+        getDrawable(context, iconResId, resSize, 0)
+
+    fun getDrawable(
+        context: Context,
+        @DrawableRes iconResId: Int,
+        resSize: Int,
+        colorRes: Int
+    ): Drawable {
+        val drawable = getDrawable(context, iconResId).mutate()
+        if (colorRes != 0) setDrawableColor(context, drawable, colorRes)
+        return setDrawableSize(context, drawable, resSize)
```

Comment:
can we do this `return if (resSize != 0) setDrawableSize(context, drawable, resSize) else drawable` when size is 0